src/main/java/org/zalando/nakadi/service/timeline/TimelineService.java

```diff
@@ -146,11 +148,18 @@ public TopicRepository getDefaultTopicRepository() throws TopicRepositoryExcepti
 
     private void switchTimelines(final Timeline activeTimeline, final Timeline nextTimeline) {
         LOG.info("Switching timelines from {} to {}", activeTimeline, nextTimeline);
-        transactionTemplate.execute(status -> {
-            try {
+        try {
+            timelineSync.startTimelineUpdate(activeTimeline.getEventType(), nakadiSettings.getTimelineWaitTimeoutMs());
+        } catch (final InterruptedException ie) {
+            Thread.currentThread().interrupt();
+            throw new TimelineException("Failed to switch timeline for:" + activeTimeline.getEventType());
+        } catch (final IllegalStateException ie) {
+            throw new ConflictException("Timeline is being created for:" + activeTimeline.getEventType(), ie);
```

Comment:
For me the message tells that Timeline is being created, which is good - because this is exactly what I wanted to do - to create a timeline :)
I think something like "Timeline is already being created for: " will make more sense;

Another very minor point - in your messages you miss a space after ":"